dazel.py

```diff
@@ -282,49 +288,57 @@ def _start_compose_services(self):
     def _run_container(self):
         """Runs the container itself."""
         print ("Starting docker container '%s'..." % self.instance_name)
-        command = "docker stop %s >/dev/null 2>&1 ; " % (self.instance_name)
-        command += "docker rm %s >/dev/null 2>&1 ; " % (self.instance_name)
-        command += "docker run -id --name=%s %s %s %s %s %s %s%s %s" % (
-            self.instance_name,
-            "--privileged" if self.docker_run_privileged else "",
-            ("-w %s" % os.path.realpath(self.directory)) if self.directory else "",
-            self.volumes,
-            self.ports,
-            ("--net=%s" % self.network) if self.network else "",
-            ("%s/" % self.repository) if self.repository else "",
-            self.image_name,
-            self.run_command if self.run_command else "")
-        command = self._with_docker_machine(command)
-        rc = os.system(command)
-        if rc:
-            return rc
+
+        try:
+            self._docker.stop(self.instance_name)
+            self._docker.remove_container(self.instance_name)
+        except docker.errors.NotFound:
+            pass
+
+        image = "%s%s" % (("%s/" % self.repository) if self.repository else "", self.image_name)
+        workdir = os.path.realpath(self.directory) if self.directory else None
+        container = self._docker.create_container(
+            image=image,
+            command=self.run_command,
+            detach=True,
+            stdin_open=True,
+            name=self.instance_name,
+            working_dir=workdir,
+            ports=list(self.ports.keys()),
+            host_config=self._docker.create_host_config(privileged=self.docker_run_privileged,
+                                                        port_bindings=self.ports,
+                                                        binds=self.volumes),
+            networking_config=self._docker.create_networking_config({
+                self.network: self._docker.create_endpoint_config()
+            }) if self.network else None,
+        )
+
+        self._docker.start(container["Id"])
 
         # Touch the dazel run file to change the timestamp.
         if self.dazel_run_file:
             open(self.dazel_run_file, "w").write(self.instance_name + "\n")
-            print ("Done.")
+        print ("Done.")
 
-        return rc
+        return 0
```

Comment:
Cause it's going to be returned from many other places, and I didn't want to refactor the whole code with that in mind. Its something we should do later on.